### ISSUE TITLE
[Bug]: useLayoutEffect does nothing on the server, because its effect cannot be encoded into the server renderer's output format

### ISSUE THREAD

**Issue by perkinsjr** [crutchcorn/houseform#17]
### Steps To Reproduce

1. Create new Next.js application (using pages directory)
2. Install Houseform
3. Add simple example 
4. Warning is presented in terminal

Full Warning: 

```
Warning: useLayoutEffect does nothing on the server, because its effect cannot be encoded into the server renderer's output format. This will lead to a mismatch between the initial, non-hydrated UI and the intended UI. 

To avoid this, useLayoutEffect should only be used in components that render exclusively on the client. See https://reactjs.org/link/uselayouteffect-ssr for common fixes.
```

### Expected Behavior

Warning shouldn't be presented to the user when using with Next.js 

### HouseForm Version

1.0.0

### React Version

React: 18.2.0  & Next.js 13.1.6

### Environment

Browser (Chrome/Firefox/Safari/etc)

**Comment by crutchcorn** [comment]
I think this should be a trivial fix by replacing our `useLayoutEffect` with a `useIsmorphicLayoutEffect` hook. Let me see if I can fix this soon

**Comment by crutchcorn** [comment]
While working on this, I noticed there are some typing and behavior issues with Zod as well - fixing that now.

**Comment by crutchcorn** [comment]
This should be fixed as-of 1.0.1, when it releases.

Heads up, if you're using TypeScript I found this other bug. It's an issue with our packaging somehow, not our TS types. I'll investigate soon.

https://github.com/crutchcorn/houseform/issues/21
